salt/states/file.py

```diff
@@ -450,7 +450,7 @@ def process_symlinks(filenames, symlinks):
             # the same list.
             _filenames = list(filenames)
             for filename in _filenames:
-                if filename.startswith(lname):
+                if filename.startswith(lname + os.sep):
```

Comment:
Can you clarify for me when a user would hit this if statement and would want to skip the file? I'm not understanding why adding the `os.sep` is correct here.